### PR TITLE
fix(structured data): drop empty columns, sanitize header, dedup header

### LINKED ISSUE #3813: [Caloris] [Structured Data] sanitize headers and dedup columns

### PR DESCRIPTION
**Pull request by fontanierh**
## Description
Fixes https://github.com/dust-tt/dust/issues/3813 and https://github.com/dust-tt/dust/issues/3814

- we want to drop the empty columns because notion DBs typically have many unused properties
- we want to slugify the column names as it limits the need to do complex escaping at query time
- we want to dedup column names as we can easily end up with several column using the same name. We'll rely on order of columns for incremental sync (🤞)

## Risk

- **NOT** backwards compatible with the current notion incremental sync. It will require a "full resync" of notion DBs structured data. This will be done in a separate PR
- Ordering of header cells becomes important if there are duplicates

### CODE REVIEW

File: front/lib/api/tables.ts [hunk 1]
```diff
@@ -431,3 +440,29 @@ async function rowsFromCsv(
...
+      acc.push(slugifiedName);
+    } else {
+      let conflictResolved = false;
+      for (let i = 2; i < 10000; i++) {
+        if (!acc.includes(slugify(`${slugifiedName}_${i}`))) {
+          acc.push(slugify(`${slugifiedName}_${i}`));
+          conflictResolved = true;
+          break;
+        }
+      }
```
Comment (flvndvd) [inline front/lib/api/tables.ts:458]: We could use a Map here to keep the current increment for each slugified header.

Comment (fontanierh) [reply front/lib/api/tables.ts:463]: My opinion here is that it wouldn't make the code more performant in any significant way and would make it overall harder to read. WDYT ?

File: front/lib/api/tables.ts [hunk 2]
```diff
@@ -331,6 +332,14 @@ async function rowsFromCsv(
     }
   }
 
+  // Drop empty columns.
+  for (const [col, values] of Object.entries(valuesByCol)) {
+    if (values.every((v) => v === "")) {
```
[the remark this answers is not in the document]
Comment (fontanierh) [reply front/lib/api/tables.ts:342]: Indeed, adding thanks

File: front/lib/api/tables.ts (hunk 1 above)
Comment (fontanierh) [reply front/lib/api/tables.ts]: reduced to 64

**Review by flvndvd — APPROVED** [review]

**Pull request merged**